extensions/pkg/controller/healthcheck/worker/helpers.go

```diff
@@ -79,41 +48,140 @@ func CheckMachineDeployment(deployment *machinev1alpha1.MachineDeployment) error
 	}
 
 	for _, trueConditionType := range trueMachineDeploymentConditionTypes {
-		conditionType := string(trueConditionType)
 		condition := getMachineDeploymentCondition(deployment.Status.Conditions, trueConditionType)
 		if condition == nil {
-			return requiredConditionMissing(conditionType)
+			return requiredConditionMissing(trueConditionType)
 		}
-		if err := general.CheckConditionState(conditionType, string(corev1.ConditionTrue), string(condition.Status), condition.Reason, condition.Message); err != nil {
+		if err := checkConditionStatus(condition, machinev1alpha1.ConditionTrue); err != nil {
 			return err
 		}
 	}
 
 	for _, trueOptionalConditionType := range trueOptionalMachineDeploymentConditionTypes {
-		conditionType := string(trueOptionalConditionType)
 		condition := getMachineDeploymentCondition(deployment.Status.Conditions, trueOptionalConditionType)
 		if condition == nil {
 			continue
 		}
-		if err := general.CheckConditionState(conditionType, string(corev1.ConditionTrue), string(condition.Status), condition.Reason, condition.Message); err != nil {
+		if err := checkConditionStatus(condition, machinev1alpha1.ConditionTrue); err != nil {
 			return err
 		}
 	}
 
 	for _, falseConditionType := range falseMachineDeploymentConditionTypes {
-		conditionType := string(falseConditionType)
 		condition := getMachineDeploymentCondition(deployment.Status.Conditions, falseConditionType)
 		if condition == nil {
 			continue
 		}
-		if err := general.CheckConditionState(conditionType, string(corev1.ConditionFalse), string(condition.Status), condition.Reason, condition.Message); err != nil {
+		if err := checkConditionStatus(condition, machinev1alpha1.ConditionFalse); err != nil {
 			return err
 		}
 	}
 
 	return nil
 }
 
+func checkMachineDeploymentsHealthy(machineDeployments []machinev1alpha1.MachineDeployment) (bool, *string, error) {
```

Comment:
Can you change the signature to return a `string` instead of `*string` as you are anyways not checking if it is `nil` and dereferencing?
```suggestion
func checkMachineDeploymentsHealthy(machineDeployments []machinev1alpha1.MachineDeployment) (bool, string, error) {
```